_posts/2024-09-19-gsoc-2024-summary-llm-hyperparameter-optimization-api.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "LLM Hyperparameter Optimization API: My Google Summer of Code Journey with Kubeflow"
+layout: post
+toc: false
+comments: true
+image: images/logo.png
+hide: false
+categories: [gsoc]
+permalink: /gsoc-2024-project-4/
+author: "Hezhi(Helen) Xie"
+---
+
+This summer, I had the opportunity to participate in the Google Summer of Code (GSoC) program, where I contributed to Kubeflow, an open-source machine learning toolkit. My project focused on developing a high-level API for optimizing hyperparameters in Large Language Models (LLMs) within Katib, Kubeflow's automated hyperparameter tuning system. I'd like to share insights from this experience with others interested in Kubeflow, GSoC, or optimizing LLMs.
+
+## Motivation
+
+The rapid advancements and rising popularity of LLMs, such as GPT and BERT, have created a growing demand for efficient LLMOps in Kubernetes. To address this, we have developed a [train API](https://www.kubeflow.org/docs/components/training/user-guides/fine-tuning/) within the Training Python SDK, simplifying the process of fine-tuning LLMs using distributed PyTorchJob workers. However, hyperparameter optimization remains a crucial yet labor-intensive task for enhancing model performance.
+
+## Goal
+
+Hyperparameter optimization is a crucial but time-consuming task in fine-tuning machine learning models, especially for LLMs that involve billions of parameters. This API aims to streamline this process by abstracting the complexity of Kubernetes infrastructure, enabling data scientists to focus on model performance instead of system configuration.
+
+![Design of API](../images/2024-09-19-gsoc-2024-llm-hyperparameter-optimization-api/design_tune_api.jpg)
+
```

Comment:
Let's also cross-reference the docs for this feature, since we will merge this PR soon: https://github.com/kubeflow/website/pull/3952